py/picca/delta_extraction/utils_pk1d.py

```diff
@@ -108,56 +110,53 @@ def exp_diff_desi(spec_dict, mask_targetid):
     Returns:
         The difference between exposures
     """
-    teff_lya = np.atleast_1d(spec_dict["TEFF_LYA"][mask_targetid])
-    argsort = np.flip(np.argsort(teff_lya))
-    flux = np.atleast_2d(spec_dict["FLUX"][mask_targetid])[argsort, :]
-    ivar = np.atleast_2d(spec_dict["IVAR"][mask_targetid])[argsort, :]
-    teff_lya = teff_lya[argsort]
+    ivar_unsorted = np.atleast_2d(spec_dict["IVAR"][mask_targetid])
+    num_exp = ivar_unsorted.shape[0]
 
-    num_exp = len(flux)
+    # Putting the lowest ivar exposure at the end if the number of exposures is odd
```

Comment:
couldn't one just inversely sort by ivar here? I.e. instead of the if statement compute `argsort=np.argsort(np.mean(ivar_unsorted,axis=1))[::-1]` and then proceed with the `flux=...` and `ivar=...` lines? Why does this need to be last? To minimize the impact of having an odd number of exp here?

Reply:
I have tested both techniques. The main problem of diff estimator is the difference between total ivar for even and odd number of exposures. If you sort by ivar, the even exposure will always have the highest total ivar which might be suboptimal. So I let the exposures like that and only put the smallest ivar in the last position.

Reply:
Ok, does that mean you basically need the first 2n exposures being in some random order to get even ivar ~ odd ivar (which might not be guaranteed to be ~random), but still want the last exposure to be the lowest ivar one (which if I read it right is then just discarded to simplify the process)?
Would it make sense to shuffle the array (once or even multiple times and averaging the diff for each shuffle) after removing the lowest ivar exp?

Reply:
Ok, given that we're currently only using diff for cross checking the pipeline noise estimates and this passes testing: lets keep it as is, we can come back to it later if we figure out that further diff calculations are needed.